src/main/java/org/example/internetshop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -101,10 +107,55 @@ public boolean delete(Long id) {
         }
     }
 
-    public Product getProductFromResultSet(ResultSet resultSet) throws SQLException {
+    @Override
+    public List<Product> getProductsByCart(Long cartId) {
+        String query = "SELECT id, product_name, price FROM shopping_carts_products scp"
+                + " JOIN products p ON scp.product_id = p.id"
+                + " WHERE scp.cart_id = ?;";
+        List<Product> products = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            setProducts(query, products, connection, cartId);
+            return products;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting products from cart failed", e);
+        }
```

Comment:
you need this method only in shoppingCartDao, so let's move it there, and you will get rid of reference between daos